keystore/v2/keystore/filesystem/export.go

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2020, Cossack Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filesystem
+
+import (
+	"errors"
+	"time"
+
+	"github.com/cossacklabs/acra/keystore/v2/keystore/api"
+	"github.com/cossacklabs/acra/keystore/v2/keystore/asn1"
+	"github.com/cossacklabs/acra/keystore/v2/keystore/crypto"
+	backendAPI "github.com/cossacklabs/acra/keystore/v2/keystore/filesystem/backend/api"
+	"github.com/cossacklabs/acra/keystore/v2/keystore/signature"
+	"github.com/cossacklabs/acra/utils"
+)
+
+// Errors returned by export/import routines.
+var (
+	ErrKeyRingExists = errors.New("imported key ring already exists")
+)
+
+func (s *KeyStore) exportKeyRings(paths []string) (rings []asn1.KeyRing, err error) {
+	rings = make([]asn1.KeyRing, len(paths))
+	defer func() {
+		if err != nil {
+			zeroizeKeyRings(rings)
+		}
+	}()
+	for i, path := range paths {
+		err := s.exportKeyRing(path, &rings[i])
+		if err != nil {
+			return nil, err
+		}
+	}
+	return rings, nil
+}
+
+func (s *KeyStore) exportKeyRing(path string, ringData *asn1.KeyRing) error {
+	ring := newKeyRing(s, path)
+	err := s.readKeyRing(ring)
+	if err != nil {
+		return err
+	}
+	*ringData, err = ring.exportASN1()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *KeyStore) importKeyRing(newRingData *asn1.KeyRing, delegate *api.KeyRingImportDelegate) error {
+	keyRing := newKeyRing(s, string(newRingData.Purpose))
+	err := s.readKeyRing(keyRing)
```

Comment:
can you add comment that here we check existing keyring with same purpose?) I wrote block of questions what we try to do until understood) because it's clear only when read block with `err == nil` below why we read empty keyring with same purpose)

Reply:
> can you add comment that here we check existing keyring with same purpose?

Will do later. Right now `importKeyRing` is more of a draft, it will be expanded in the next PR. I hope it will be more readable then.